From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Max Lee <yowching.lee@gmail.com>
Date: Sun, 10 May 2026 21:45:00 +0800
Subject: [PATCH] fix aarch64 cfi startproc placement on darwin

Apple's arm64 assembler rejects CFI state changes when .cfi_startproc
is emitted before the function label.  Keep the same CFI coverage, but
start the CFI procedure after the symbol label for the two affected
entry points.
---
 src/aarch64/sysv.S | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/aarch64/sysv.S b/src/aarch64/sysv.S
index fdd0e8b..bbda221 100644
--- a/src/aarch64/sysv.S
+++ b/src/aarch64/sysv.S
@@ -88,9 +88,8 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
    x4 flags
    x5 closure
 */
-
-	cfi_startproc
 CNAME(ffi_call_SYSV):
+	cfi_startproc
 	BTI_C
 	/* Sign the lr with x1 since that is where it will be stored */
 	SIGN_LR_WITH_REG(x1)
@@ -347,8 +346,8 @@ CNAME(ffi_closure_SYSV_V):
 #endif
 
 	.align	4
-	cfi_startproc
 CNAME(ffi_closure_SYSV):
+	cfi_startproc
 	BTI_C
 	SIGN_LR
 	stp     x29, x30, [sp, #-ffi_closure_SYSV_FS]!
-- 
2.50.1

